api/controllers/calendarEntry.controller.ts

```diff
@@ -43,82 +39,46 @@ type RecurringParentEntry = {
   startTimeUtc: Date;
   endTimeUtc: Date;
   frequency: string;
-  recurrenceBegins: Date;
-  recurrenceEnds: Date;
-  createdAt: Date;
-  updatedAt: Date;
-};
-
-type RecurringChildEntry = {
-  id: string;
-  eventId: string;
-  creatorId: string;
-  title: string;
-  description?: string;
-  allDay: boolean;
-  recurring: boolean;
-  frequency: string;
-  recurrenceBegins: Date;
-  recurrenceEnds: Date;
-  startTimeUtc: Date;
-  endTimeUtc: Date;
-  recurringEventId: mongoose.Schema.Types.ObjectId;
+  recurrenceEndsUtc: Date;
+  recurrencePattern: string;
   createdAt: Date;
   updatedAt: Date;
 };
 
+// not used right now but will be used in the future
 const isNonRecurringEntry = (entry) => {
   return (entry as NonRecurringEntry).recurring === false;
 };
 
-const isRecurringParentEntry = (entry) => {
-  return (
-    (entry as RecurringParentEntry).recurring === true &&
-    entry.recurringEventId === undefined
-  );
+const isRecurringEntry = (entry) => {
+  return (entry as RecurringEntry).recurring === true;
```

Comment:
I _think_ you can simplify this check using `instanceof` and maybe delete this function altogether and just use that wherever you need it ie `entry instanceof RecurringEntry` rather than `isRecurringEntry(entry)`

Reply:
This conforms to the documentation and would prefer to keep it as is.
[docs:](https://www.typescriptlang.org/docs/handbook/2/narrowing.html)

`function isFish(pet: Fish | Bird): pet is Fish {
  return (pet as Fish).swim !== undefined;
}`